x-pack/plugins/apm/server/lib/apm_telemetry/collect_data_telemetry/tasks.ts

```diff
@@ -1082,6 +1093,79 @@ export const tasks: TelemetryTask[] = [
   {
     name: 'indices_stats',
     executor: async ({ indices, telemetryClient }) => {
+      const dsRollupDictionary: DataStreamWithRollup =
+        {} as DataStreamWithRollup;
+      const dsWithoutRollupDictionary: DataStreamWithoutRollup =
+        {} as DataStreamWithoutRollup;
+
+      const metricSetsSupportingRollUps: MetricSupportingRollUp[] = [
+        MetricTypes.service_destination,
+        MetricTypes.transaction,
+        MetricTypes.service_summary,
+        MetricTypes.service_transaction,
+        MetricTypes.span_breakdown,
+      ];
+
+      const metricSetsNotSupportingRollUps: MetricNotSupportingRollup[] = [
+        MetricTypes.app,
+      ];
+
+      const rollUpIntervals: MetricRollupIntervals[] = [
+        RollupInterval.OneMinute,
+        RollupInterval.TenMinutes,
+        RollupInterval.SixtyMinutes,
+      ];
+
+      // The API calls must be done in series rather than in parallel due to the nature
+      // of how tasks are executed. We don't want to burden the customers instances
+      // which could directly impact the performance on the UI.
+      const fetchRollupMetrics = async () => {
+        for (const metricSet of metricSetsSupportingRollUps) {
+          for (const bucketSize of rollUpIntervals) {
+            const datastream = `metrics-apm.${metricSet}.${bucketSize}-default*`;
+            const response = await telemetryClient.indicesStats({
+              index: [datastream],
+              expand_wildcards: 'all',
+              filter_path: ['_all', '_shards'],
+            });
+            dsRollupDictionary[metricSet] = dsRollupDictionary[metricSet] || {};
+            dsRollupDictionary[metricSet][bucketSize] = {
+              total: {
+                shards: response?._shards?.total ?? 0,
+                docs: { count: response?._all?.total?.docs?.count ?? 0 },
```

Comment:
@dgieselaar @achyutjhunjhunwala do you think it makes more sense to get docs from all shards or just primaries?

Reply:
primaries only

Reply:
That's what I thought, in this case this should be changed to `_all.primaries` instead and update the metadata description accordingly.

Reply:
Makes sense. 👍🏼